yarn-project/foundation/src/serialize/types.ts

```diff
@@ -16,7 +16,7 @@ type _Tuple<T, N extends number, R extends unknown[]> = R['length'] extends N ?
  */
 export function assertLength<T, N extends number>(array: T[], n: N): Tuple<T, N> {
   if (array.length !== n) {
-    throw new Error("Wrong 'fixed array' size");
+    throw new Error(`Wrong 'fixed array' size. Expected ${n}, got ${array.length}.`);
```

Comment:
Whenever I see a change like this one I wonder how many hours of debugging were lost to lack of meaningful error messages.